python/grass/__init__.py

```diff
@@ -75,16 +93,48 @@ def _translate(text):
             # does not raise an exception even if the locale settings is broken
             # or the translation files were not installed.
             fallback = True
-            translation = gettext.translation(
-                "grasslibs", locale_dir, fallback=fallback
-            )
-            # Add other domains as fallback.
-            translation.add_fallback(
-                gettext.translation("grassmods", locale_dir, fallback=fallback)
-            )
-            translation.add_fallback(
-                gettext.translation("grasswxpy", locale_dir, fallback=fallback)
-            )
+            translation = None
+            domains = ("grasslibs", "grassmods", "grasswxpy")
+
+            def apply_trans(kwargs):
+                nonlocal translation
+                if kwargs["domain"] == "grasslibs":
+                    translation = gettext.translation(**kwargs)
+                else:
+                    translation.add_fallback(gettext.translation(**kwargs))
+
+            try:
+                # Get user defined lang
+                lang = _UserSettings.Get(
+                    group="language",
+                    key="locale",
+                    subkey="lc_all",
+                )
+            except NameError:
+                # If _UserSettings isn't imported during tests (without wxGUI)
+                lang = None
+
+            if lang and lang not in ("en", "system"):
+                for domain in domains:
+                    apply_trans(
+                        {
+                            "domain": domain,
+                            "localedir": locale_dir,
+                            "fallback": fallback,
+                            "languages": (lang,),
```

Comment:
This can probably become something like `languages=[lang] if lang else None`:

* [gettext](https://docs.python.org/3.10/library/gettext.html#gettext.find) doc says it is a list: _Optional languages is a list of strings, where each string is a language code._ (Probably doesn't matter much, but increasingly popular practice is to have things which are lists as lists even when they are fixed size.)
* You can resolve any language assignment special needs/reassignments before the apply_translation/gettext.translation call.
* The dict syntax should no be needed if kwargs is handled fully or if gettext.translation is called directly.

Reply:
Yes, you're right.